spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -279,6 +301,205 @@ public void init() throws Exception {
+			InetAddress result = null;
+
+			try {
+				int lowest = 2147483647;
+				Enumeration nics = NetworkInterface.getNetworkInterfaces();
+
+				label61:
+				while(true) {
+					NetworkInterface ifc;
+					do {
+						while(true) {
+							do {
+								if (!nics.hasMoreElements()) {
+									break label61;
+								}
+
+								ifc = (NetworkInterface)nics.nextElement();
+							} while(!ifc.isUp());
+
+							this.log.trace("Testing interface: " + ifc.getDisplayName());
+							if (ifc.getIndex() >= lowest && result != null) {
+								if (result != null) {
+									continue;
+								}
+								break;
+							}
+
+							lowest = ifc.getIndex();
+							break;
+						}
+					} while(this.ignoreInterface(ifc.getDisplayName()));
+
+					Enumeration addrs = ifc.getInetAddresses();
+
+					while(addrs.hasMoreElements()) {
+						InetAddress address = (InetAddress)addrs.nextElement();
+						if (address instanceof Inet6Address && !address.isLoopbackAddress() && this.isPreferredAddress(address)) {
+							this.log.trace("Found non-loopback interface: " + ifc.getDisplayName());
+							result = address;
+						}
+					}
+				}
+			} catch (IOException var8) {
+				this.log.error("Cannot get first non-loopback address", var8);
+			}
+
+			if (result != null) {
+				return result;
+			} else {
+				try {
+					return InetAddress.getLocalHost();
+				} catch (UnknownHostException var7) {
+					this.log.warn("Unable to retrieve localhost");
+					return null;
+				}
+			}
+		}
+
+		 boolean isPreferredAddress(InetAddress address) {
+			 if (this.isUseOnlySiteLocalInterfaces()) {
+				 boolean siteLocalAddress = address.isSiteLocalAddress();
+				 if (!siteLocalAddress) {
+					 this.log.trace("Ignoring address: " + address.getHostAddress());
+				 }
+
+				 return siteLocalAddress;
+			 } else {
+				 List<String> preferredNetworks = this.getPreferredNetworks();
+				 if (preferredNetworks.isEmpty()) {
+					 return true;
+				 } else {
+					 Iterator var3 = preferredNetworks.iterator();
+
+					 String regex;
+					 String hostAddress;
+					 do {
+						 if (!var3.hasNext()) {
+							 this.log.trace("Ignoring address: " + address.getHostAddress());
+							 return false;
+						 }
+
+						 regex = (String)var3.next();
+						 hostAddress = address.getHostAddress();
+					 } while(!hostAddress.matches(regex) && !hostAddress.startsWith(regex));
+
+					 return true;
+				 }
+			 }
+		 }
+
+		 boolean ignoreInterface(String interfaceName) {
+			 Iterator var2 = this.getIgnoredInterfaces().iterator();
+
+			 String regex;
+			 do {
+				 if (!var2.hasNext()) {
+					 return false;
+				 }
+
+				 regex = (String)var2.next();
+			 } while(!interfaceName.matches(regex));
+
+			 this.log.trace("Ignoring interface: " + interfaceName);
+			 return true;
+		 }
+
+		 public InetUtils.HostInfo convertAddress(final InetAddress address) {
+			 InetUtils.HostInfo hostInfo = new InetUtils.HostInfo();
+			 ExecutorService var10000 = this.executorService;
```

Comment:
Don't use the variable like var10000, tt looks very random, similar problems in other places need to be fixed